tree/tree/src/TBranch.cxx

```diff
@@ -2036,6 +2047,22 @@ TBranch* TBranch::GetMother() const
 {
    if (fMother) return fMother;
 
+   {
+      TBranch *parent = fParent;
+      while(parent) {
+         if (parent->fMother) {
+            const_cast<TBranch*>(this)->fMother = parent->fMother; // We can not yet use the 'mutable' keyword
```

Comment:
That breaks any MT guarantee. Can you create a non-const overload that does the caching of `fMother`?

Reply:
Yes that is a problem.

Reply:
> Can you create a non-const overload that does the caching of fMother?

I could but it would defy the purpose.  The non-const version would have un-predictable performance (Slow until a non-const call to GetMother in the cases where fMother is not set at construction).

I.e. I need to make it thread safe ...